src/main.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Display.c
+ *
+ *  Created on: 14.08.2017
+ *      Author: darek
+ */
+// #include <driver/i2c.h>
+#include <esp_err.h>
+#include <esp_log.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+#include <esp_system.h>
+#include "imu.h"
+#include "sdkconfig.h"
+#include <cstring>
+#include <array>
+#include <string.h>
+#include "driver/uart.h"
+#include <driver/ledc.h>
+#include <esp32/gpio.h>
+
+static const int RX_BUF_SIZE = 1024;
+
+#define TXD_PIN (GPIO_NUM_17)
+#define RXD_PIN (GPIO_NUM_16)
+
+#define PACKET_SIZE 16
+const int DELAY_MS = 30;
+
+extern "C" {
+  void app_main(void);
+}
+
+// MOTOR CONSTANTS
+const int PWM_CHANNEL = 0;
+const int PWM_FREQ = 50000;
+const int PWM_RESOLUTION = 8;
+const float NO_LOAD_SPEED = 7.50f;
+const float STALL_TORQUE = 1.05f;
+
+struct Motor {
+  gpio_num_t encoder;  // yellow
+  gpio_num_t pwm;// blue
+  gpio_num_t dir; // white
+};
+
+const Motor motor1 = { GPIO_NUM_18, GPIO_NUM_5, GPIO_NUM_19 };   // left
+const Motor motor2 = { GPIO_NUM_12, GPIO_NUM_14, GPIO_NUM_27 };   // right
+
+
+static IMUData imu_data;
+
+
+void init_uart(void);
+void init_motors(void);
+void imu_cb(IMUData data) { imu_data = data; }
+void tx_task(void); // read imu_data and send it over uart
+void set_torque(float torque);
+
+
+void app_main() {
+  init_uart();
+  // xTaskCreate(task_initI2C, "task_initI2C", 2048, NULL, 5, NULL);
+  xTaskCreate(tx_task, "task_display", 4096, NULL, 5, NULL);
+}
+
+
+
+static void rx_task(void* arg) {
+  static const char* RX_TASK_TAG = "RX_TASK";
+  esp_log_level_set(RX_TASK_TAG, ESP_LOG_INFO);
+  uint8_t* data = (uint8_t*)malloc(RX_BUF_SIZE + 1);
```

Comment:
We can make this into an array of size `RX_BUF_SIZE` to avoid the allocations, no?

Reply:
I'll remove it.

Reply:
Okayy!